composeApp/src/commonMain/kotlin/io/github/couchtracker/ui/components/DateTimePickerDialog.kt

```diff
@@ -0,0 +1,294 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDefaults
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.DatePickerState
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.SegmentedButton
+import androidx.compose.material3.SegmentedButtonDefaults
+import androidx.compose.material3.SingleChoiceSegmentedButtonRow
+import androidx.compose.material3.SingleChoiceSegmentedButtonRowScope
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.TimePicker
+import androidx.compose.material3.TimePickerState
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.material3.rememberTimePickerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import couch_tracker_app.composeapp.generated.resources.Res
+import couch_tracker_app.composeapp.generated.resources.back_action
+import couch_tracker_app.composeapp.generated.resources.cancel_action
+import couch_tracker_app.composeapp.generated.resources.continue_action
+import couch_tracker_app.composeapp.generated.resources.date_and_time_dialog_precision
+import couch_tracker_app.composeapp.generated.resources.date_and_time_precision_date
+import couch_tracker_app.composeapp.generated.resources.date_and_time_precision_month
+import couch_tracker_app.composeapp.generated.resources.date_and_time_precision_year
+import couch_tracker_app.composeapp.generated.resources.skip_action
+import io.github.couchtracker.db.user.model.partialtime.PartialDateTime
+import io.github.couchtracker.intl.STRING_OK
+import io.github.couchtracker.intl.datetime.Skeletons
+import io.github.couchtracker.intl.datetime.localized
+import io.github.couchtracker.utils.str
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.LocalTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
+
+private enum class DatePrecision {
+    YEAR,
+    MONTH,
+    DATE,
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Suppress("LongMethod")
+@Composable
+fun DateTimePickerDialog(
+    initialDate: PartialDateTime.Local?,
+    showTimePicker: Boolean,
+    openTimePicker: (Boolean) -> Unit,
```

Comment:
Suggestion: these two parameters are not very clear.

- `showTimePicker`: an enum with two states `DATE` and `TIME` would be clearer. Could be called `pickerType`
- `openTimePicker`: this is even more confusing. Why is it accepting a Boolean? Are we ever passing `false` to this callback? Given the change above, it could be called `onPickerTypeChange`